src/test/groovy/org/prebid/server/functional/tests/GppCookieSyncSpec.groovy

```diff
@@ -0,0 +1,260 @@
+            "adapters.${RUBICON.value}.enabled"                    : "true",
+            "adapters.${RUBICON.value}.usersync.cookie-family-name": RUBICON.value,]
+    private static final Map<String, String> APPNEXUS_CONFIG = [
+            "adapters.${APPNEXUS.value}.enabled"                    : "true",
+            "adapters.${APPNEXUS.value}.usersync.cookie-family-name": APPNEXUS.value]
+    private static final Map<String, String> PBS_CONFIG = APPNEXUS_CONFIG + RUBICON_CONFIG + GENERIC_CONFIG
+
+    private PrebidServerService prebidServerService = pbsServiceFactory.getService(PBS_CONFIG)
+
+    def "PBS cookie sync request should reflect error when GPP is specified, but not comma-separate and gdpr is empty"() {
+        given: "Request without GDPR and invalid GPP"
+        def gppSid = "${PBSUtils.randomNumber}${PBSUtils.randomNumber}"
+        def cookieSyncRequest = CookieSyncRequest.defaultCookieSyncRequest.tap {
+            it.gpp = null
+            it.gppSid = gppSid
+            it.gdpr = null
+        }
+
+        when: "PBS processes cookie sync request"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should contain error"
+        def bidderStatus = response.getBidderUserSync(GENERIC)
+        assert bidderStatus.error == "Invalid GPP value" // TODO replace with error message
+
+        and: "Metric should contain cookie_sync.FAMILY.tcf.blocked"
+        def metric = this.prebidServerService.sendCollectedMetricsRequest()
+        assert metric["cookie_sync.generic.tcf.blocked"] == 1
+    }
+
+    def "PBS cookie sync request should set GBPR to 1 when GPP contains 2"() {
+        given: "Request without GDPR and invalid GPP"
+
+        def gppSid = "${PBSUtils.randomNumber},${PBSUtils.randomNumber}, 2"
+        def cookieSyncRequest = CookieSyncRequest.defaultCookieSyncRequest.tap {
+            it.bidders = []
+            it.gpp = null
+            it.gppSid = gppSid
+            it.gdpr = null
+        }
+
+        when: "PBS processes cookie sync request"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+
+        then: "Response should contain gdpr as 1"
+        assert response.gdpr == 1
+    }
+
+    def "PBS cookie sync request should set GBPR to 0 when GPP contains not 2"() {
+        given: "Request without GDPR and invalid GPP"
+
+        def gppSid = "0, 1"
+        def cookieSyncRequest = CookieSyncRequest.defaultCookieSyncRequest.tap {
+            it.bidders = []
+            it.gpp = null
+            it.gppSid = gppSid
+            it.gdpr = null
+        }
+
+        when: "PBS processes cookie sync request"
+        def response = prebidServerService.sendCookieSyncRequest(cookieSyncRequest)
+        then: "Response should contain gdpr as 0"
+        assert response.gdpr == 0
+    }
+
+    def "PBS cookie sync request should reflect warning when GPP is contains 2  and gdpr is not 1"() {
```

Comment:
1) "reflect" -> "respond with"
2) "PBS cookie sync request should respond with a warning when GPP SID contains 2 and gdpr is not 1"